woocommerce/woocommerce
# Add order source attribution to the checkout block

This is addition to https://github.com/woocommerce/woocommerce/pull/39701

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:



- [x] Move JS changes...
- [x] Move PHP changes from blocks repo PR
	https://github.com/woocommerce/woocommerce-blocks/pull/11506 
- [x] Remove jQuery dependency from the checkout frontend
	
	
![image](https://github.com/woocommerce/woocommerce/assets/17435/977d01a7-c856-4743-bbf1-0011d1b87265)



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Checkout the branch and build
2. Add items to the cart
3. go the the block-based checkout
4. Open network tab in the dev tools
5. "Place order"
5. Check the presence and contents of `extensions["woocommerce/order-source-attribution"]` in the payload of `/wp-json/wc/store/v1/checkout` request
6. Check the order metadata is saved correctly.



### Changelog entry




<details>

#### Significance




#### Type




#### Message 

#### Comment 

</details>

### Additional notes
1. I added the JS file too the <code>/client/<strong>legacy</strong></code> even though. It's a piece of vanilla JS targeting the **modern** blocks. But I didn't find a better place.
